ReactiveSwift-UIExamples.playground/Pages/ValidatingProperty.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,149 @@
+import ReactiveSwift
+import Result
+import UIKit
+import PlaygroundSupport
+
+final class ViewModel {
+	struct FormError: Error {
+		let reason: String
+
+		static let invalidEmail = FormError(reason: "The address must end with `@reactivecocoa.io`.")
+		static let mismatchEmail = FormError(reason: "The e-mail addresses do not match.")
+		static let usernameUnavailable = FormError(reason: "The username has been taken.")
+	}
+
+	let email: MutableValidatingProperty<String, FormError>
+	let emailConfirmation: MutableValidatingProperty<String, FormError>
+	let termsAccepted: MutableProperty<Bool>
+
+	let reasons: Signal<String, NoError>
+
+	let submit: Action<(), (), FormError>
+
+	init(userService: UserService) {
+		email = MutableValidatingProperty<String, FormError>("") { input in
+			return input.hasSuffix("@reactivecocoa.io") ? .success : .failure(.invalidEmail)
+		}
+
+		emailConfirmation = MutableValidatingProperty<String, FormError>("", with: email) { input, email in
+			return input == email ? .success : .failure(.mismatchEmail)
+		}
+
+		termsAccepted = MutableProperty(false)
+
+		// Aggregate latest failure contexts as a stream of strings.
+		reasons = Property.combineLatest(email.result, emailConfirmation.result)
+			.signal
+			.debounce(0.1, on: QueueScheduler.main)
+			.map { [$0, $1].flatMap { $0.error?.reason }.joined(separator: "\n") }
+
+		// A `Property` of the validated username.
+		//
+		// It outputs the valid username for the `Action` to work on, or `nil` if the form
+		// is invalid and the `Action` would be disabled consequently.
+		let validatedEmail = Property.combineLatest(email.result,
+		                                            emailConfirmation.result,
+		                                            termsAccepted)
+			.map { !$0.isFailure && !$1.isFailure && $2 }
+			.combineLatest(with: email)
```

Comment:
I don't actually think this is safe. `email` here isn't guaranteed to be the same one that passed the check on the line before. I think you need to read the value and the value portion together.